src/direct_mlj.jl

```diff
@@ -0,0 +1,787 @@
+            end
+
+            # fit the Laplace model:
+            LaplaceRedux.fit!(la, data_loader)
+            optimize_prior!(la; verbose=false, n_steps=m.fit_prior_nsteps)
+
+            fitresult = (la, decode)
+            report = (loss_history=old_loss_history,)
+            cache = (deepcopy(m), old_state_tree, old_loss_history)
+
+        else
+            println(
+                "The number of epochs inserted is lower than the number of epochs already been trained. No update is necessary",
+            )
+            fitresult = (old_la, decode)
+            report = (loss_history=old_loss_history,)
+            cache = (deepcopy(m), old_state_tree, old_loss_history)
+        end
+
+    elseif MMI.is_same_except(
+        m,
+        old_model,
+        :fit_prior_nsteps,
+        :subset_of_weights,
+        :subnetwork_indices,
+        :hessian_structure,
+        :backend,
+        :σ,
+        :μ₀,
+        :P₀,
+    )
+        println(" updating only the laplace optimization part")
+        old_la = old_fitresult[1]
+
+        la = LaplaceRedux.Laplace(
+            old_la.model;
+            likelihood=:regression,
+            subset_of_weights=m.subset_of_weights,
+            subnetwork_indices=m.subnetwork_indices,
+            hessian_structure=m.hessian_structure,
+            backend=m.backend,
+            σ=m.σ,
+            μ₀=m.μ₀,
+            P₀=m.P₀,
+        )
+        if typeof(m) == LaplaceClassifier
+            la.likelihood = :classification
+        end
+
+        # fit the Laplace model:
+        LaplaceRedux.fit!(la, data_loader)
+        optimize_prior!(la; verbose=false, n_steps=m.fit_prior_nsteps)
+
+        fitresult = (la, decode)
+        report = (loss_history=old_loss_history,)
+        cache = (deepcopy(m), old_state_tree, old_loss_history)
+
+    else
+        fitresult, cache, report = MLJBase.fit(m, verbosity, X, y)
+    end
+
+    return fitresult, cache, report
+end
+
+@doc """
+    function MMI.is_same_except(m1::LaplaceModels, m2::LaplaceModels, exceptions::Symbol...) 
+
+If both `m1` and `m2` are of `MLJType`, return `true` if the
+following conditions all hold, and `false` otherwise:
+
+- `typeof(m1) === typeof(m2)`
+
+- `propertynames(m1) === propertynames(m2)`
+
+- with the exception of properties listed as `exceptions` or bound to
+  an `AbstractRNG`, each pair of corresponding property values is
+  either "equal" or both undefined. (If a property appears as a
+  `propertyname` but not a `fieldname`, it is deemed as always defined.)
+
+The meaining of "equal" depends on the type of the property value:
+
+- values that are themselves of `MLJType` are "equal" if they are
+  equal in the sense of `is_same_except` with no exceptions.
+
+- values that are not of `MLJType` are "equal" if they are `==`.
+
+In the special case of a "deep" property, "equal" has a different
+meaning; see [`MMI.StatTraits.deep_properties`](@ref)) for details.
```

Comment:
The problem was that Documenter couldn't find this reference without the namespace.

Reply:
@pat-alt  can you explain me how to use that "@ref" ? what does it mean? i guess it is used for crossreferencing but i have never used it